### ISSUE TITLE
yAxis min max change when browser/div resizes

### ISSUE THREAD

**Issue by dereklin** [highcharts/highcharts#7813]
#### Expected behaviour
yAxis min and max should stay the same when browser/div resizes
![highcharts-correct-yaxis-min-max](https://user-images.githubusercontent.com/2231799/35889141-325150ec-0b4f-11e8-9193-fed126e62f33.PNG)

#### Actual behaviour
yAxis min and max change.  Chart doesn't look good
![highcharts-wrong-yaxis-min-max](https://user-images.githubusercontent.com/2231799/35889116-109495ae-0b4f-11e8-8fab-14294c6a0135.PNG)

#### Live demo with steps to reproduce
http://jsfiddle.net/dereklin/Lkthktpf/14/
Make the browser pane smaller to see the yAxis min max change

#### Product version
https://code.highcharts.com/highcharts.js
current version

#### Affected browser(s)
Chrome

I tried different ways to fix the yAxis min/max.  Could you provide a work around?

**Comment by pawelfus** [comment]
Hi @dereklin 

Thank you for reporting the issue. This is interesting idea of translating values in tooltip/labels. The same can be achieved using "original" data, with `series.threshold` option, take a look: http://jsfiddle.net/BlackLabel/Lkthktpf/15/

Regarding your demo, it's caused by [`yAxis.endOnTick`](https://api.highcharts.com/highcharts/yAxis.endOnTick) (and `startOnTick`), you can simply disable these options: http://jsfiddle.net/BlackLabel/Lkthktpf/19/

It's explained in [`yAxis.max`](https://api.highcharts.com/highcharts/yAxis.max) (and `min`) options:

> If the endOnTick option is true, the max value might be rounded up.

**Comment by dereklin** [comment]
@pawelfus 
Thanks for showing me series.threshold!
It seems like I don't even need to set endOnTick/startOnTick to false when I use the "original" data.  I am guess that's because there is no more calculations?

**Comment by pawelfus** [comment]
Let's use your demo **without** `yAxis.labels.formatter`: http://jsfiddle.net/BlackLabel/Lkthktpf/22/ - as you can see, tickInterval=20 and first tick = 60. If we start with 50, then we won't show 0 (which in general, should be rendered on chart, right? :) ), so we use `startOnTick` and `endOnTick` to add extra space.

**Comment by dereklin** [comment]
@pawelfus 
I wasn't aware of these nuances before.  It looks like when the div size is big, the tickInterval is calculated to be 10, and that's why I don't see the problem.

When the div is small, the tickInterval is 20, and the starting tick is -60.  I still don't understand why the starting tick can't stick to -50 (the min)...

**Comment by pawelfus** [comment]
@dereklin 
> If we start with 50, then we won't show 0 (which, in general, should be rendered on chart, right? :) ), so we use startOnTick and endOnTick to add extra space.

**Comment by dereklin** [comment]
@pawelfus 
My original code starts at -50 and ends at 50. (min/max)  If the tickeInterval is 20, it can be:
-50, -30, -10, 10, 30, 50

I think what you are implying is that 0 is an important number, like a default starting point, and highcharts is trying its best to show 0; and therefore, it changes the min to -60 so that 0 will show up.

Is that right?

**Comment by pawelfus** [comment]
tl;dr Yes :)

Long story: In general `0` in charting is an important number as we always start from that value :) -50 is changed to -60 because of `startOnTick`. Maybe something is misinterpreted here about `min` and `max`. `min` and `max` options don't set the first/last tick on the chart (not always at least), but **extremes** on an xAxis. First tick is calculated as `Math.floor(min / tickInterval) * tickInterval`, in your case: `Math.floor(-50 / 20) * 20 = -60` (and the same way the last tick, equals `60`). Now we fill ticks: `-60, -40, -20, 0, 20, 40, 60` and we compare calculated ticks with `start/endOnTick` and `min/max` options (and many other options ;) ). If `min == -50`, and first tick doesn't equal min => go check `startOnTick`, if enabled, add extra space and render -60, if disabled don't render -60. So it's not like we hardcoded somewhere `0` as a special number, but all options/algorithms behaves like 0 is special (which works fo most of the datasets that cross the origin point). Of course, it's not like you can't change that, we are aware that for some datasets user may need very specific algorithm, and one can change how ticks are calculated using [`yAxis.tickPositioner`](https://api.highcharts.com/highcharts/yAxis.tickPositioner).
